File: src/me/krickl/memebotj/InternalCommands/AdminCommands/JoinCommand.scala
package me.krickl.memebotj.InternalCommands.AdminCommands

import me.krickl.memebotj.{ChannelHandler, CommandHandler, Memebot, UserHandler}
import scala.collection.JavaConversions._

class JoinCommand(channel: String, command: String, dbprefix: String) extends CommandHandler(channel,
	command, dbprefix) {

	this.setAccess("botadmin")

	this.setNeededCommandPower(10)

	this.setHelptext(Memebot.formatText("JOIN_SYNTAX", channelOriginHandler, null, this, true, Array()))

	override def commandScript(sender: UserHandler, channelHandler: ChannelHandler, data: Array[String]) {
		try {
			if(CommandHandler.checkPermission(sender.getUsername, 75, channelHandler.userList) && data.length > 1) {
        Memebot.joinChannel("#" + data(0))
        channelHandler.sendMessage(Memebot.formatText(channelHandler.localisation.localisedStringFor("JOIN"), channelHandler, sender, this, false, Array(data(0))), this.getChannelOrigin)
      } else if(channelHandler.channel == Memebot.mainChannel) {
        for(channel <- Memebot.joinedChannels) {
          if(channel.channel == "#" + sender.getUsername) {
            channelHandler.sendMessage(Memebot.formatText(channelHandler.localisation.localisedStringFor("JOIN_FAIL"), channelHandler, sender, this, false, Array(sender.getUsername)), this.getChannelOrigin)
            return
          }
        }
        Memebot.joinChannel("#" + sender.getUsername)
        channelHandler.sendMessage(Memebot.formatText(channelHandler.localisation.localisedStringFor("JOIN"), channelHandler, sender, this, false, Array(sender.getUsername)), this.getChannelOrigin)
      }
		} catch {
			case e: ArrayIndexOutOfBoundsException =>
		}
	}
}

File: src/main/java/me/krickl/memebotj/Commands/Internal/JoinCommand.java
package me.krickl.memebotj.Commands.Internal;

import me.krickl.memebotj.Channel.ChannelHandler;
import me.krickl.memebotj.Commands.CommandHandler;
import me.krickl.memebotj.Memebot;
import me.krickl.memebotj.User.UserHandler;

/**
 * This file is part of memebotj.
 * Created by unlink on 11/04/16.
 */
public class JoinCommand extends CommandHandler {
    public JoinCommand(ChannelHandler channelHandler, String commandName, String dbprefix) {
        super(channelHandler, commandName, dbprefix);
    }

    @Override
    public void overrideDB() {
        this.setNeededCommandPower(10);

        this.setHelptext(Memebot.formatText("JOIN_SYNTAX", getChannelHandler(), null, this, true, new String[]{}, ""));
    }

    @Override
    public void commandScript(UserHandler sender, String[] data) {
        try {
            for (ChannelHandler channel : Memebot.joinedChannels) {
                if (channel.getChannel().equals("#" + sender.getUsername().toLowerCase())) {
                    getChannelHandler().sendMessage(
                            Memebot.formatText(getChannelHandler().getLocalisation().localisedStringFor("JOIN_FAIL"),
                                    getChannelHandler(), sender, this, false, new String[]{sender.getUsername()}, ""),
                            getChannelHandler().getChannel(), sender, isWhisper());
                    return;
                }
            }
            Memebot.joinChannel("#" + sender.getUsername());
            getChannelHandler().sendMessage(Memebot.formatText(getChannelHandler().getLocalisation().localisedStringFor("JOIN"),
                    getChannelHandler(), sender, this, false, new String[]{sender.getUsername()}, ""), getChannelHandler().getChannel(),
                    sender, isWhisper());
        } catch (ArrayIndexOutOfBoundsException e) {
            e.printStackTrace();
        }
    }
}
